contracts/interchain-token-service/src/contract/execute/mod.rs

```diff
@@ -391,39 +410,46 @@ pub fn enable_execution(deps: DepsMut) -> Result<Response, Error> {
     killswitch::disengage(deps.storage, Event::ExecutionEnabled).change_context(Error::State)
 }
 
-pub fn register_chains(deps: DepsMut, chains: Vec<msg::ChainConfig>) -> Result<Response, Error> {
+pub fn register_chains(
+    mut deps: DepsMut,
+    chains: Vec<msg::ChainConfig>,
+) -> Result<Response, Error> {
     chains
         .into_iter()
-        .try_for_each(|chain_config| register_chain(deps.storage, chain_config))?;
+        .try_for_each(|chain_config| register_chain(&mut deps, chain_config))?;
 
     Ok(Response::new())
 }
 
-fn register_chain(storage: &mut dyn Storage, config: msg::ChainConfig) -> Result<(), Error> {
-    match state::may_load_chain_config(storage, &config.chain).change_context(Error::State)? {
-        Some(_) => bail!(Error::ChainAlreadyRegistered(config.chain)),
-        None => state::save_chain_config(storage, &config.chain.clone(), &config.into())
+fn register_chain(deps: &mut DepsMut, config: msg::ChainConfig) -> Result<(), Error> {
+    let chain = config.chain.clone();
+    let validated_config =
```

Comment:
```suggestion
    let config =
```
shadow the old one so it can't be accessed anymore? also the more idiomatic pattern in Rust